### PR TITLE
feat: use links instead of divs for badges

### PR DESCRIPTION
**Pull request by braianj**
The CategoryBadge, GenderBadge and SmartBadge components were changed to create the url and send it to the IconBadge component.
IconBadge was also modified to accept the url.
This is related to #622

### CODE REVIEW

File: webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx [hunk 1]
```diff
@@ -2,13 +2,32 @@ import React from 'react'
...
+
+  const href = React.useMemo(() => {
+    let browseProps = { assetType: assetType, section: section }
+    !section && delete browseProps.section
+
+    return locations.browse(browseProps)
+  }, [assetType, section])  
+
+  // TODO: we have to handle these types of errors and report them somewhere
+  React.useEffect(() => {
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx:20]: We could just import `useEffect` right?

```suggestion
  useEffect(() => {
```

File: webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx [hunk 2]
```diff
@@ -2,13 +2,32 @@ import React from 'react'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 import IconBadge from '../IconBadge'
 import { Props } from './CategoryBadge.types'
+import { getSearchWearableSection } from '../../../modules/routing/search'
+import { locations } from '../../../modules/routing/locations'
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx:6]: We have some (annoying maybe :P) [import order standards](https://www.notion.so/decentraland/dApps-Standards-760ee6551c9f4c00aa71f7d9c50a7646#8d48e37938124a41ac664112cb1e27f6) we should follow!

```suggestion
import { t } from 'decentraland-dapps/dist/modules/translation/utils'
import { getSearchWearableSection } from '../../../modules/routing/search'
import { locations } from '../../../modules/routing/locations'
import IconBadge from '../IconBadge'
import { Props } from './CategoryBadge.types'
```

File: webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx [hunk 3]
```diff
@@ -2,13 +2,32 @@ import React from 'react'
...
+
+const CategoryBadge = ({ wearable, assetType }: Props) => {
+
+  const section = getSearchWearableSection(wearable.category)
+
+  const href = React.useMemo(() => {
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx:12]: We could just import `useMemo` right?

```suggestion
  useMemo(() => {
```

File: webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx [hunk 4]
```diff
@@ -2,13 +2,32 @@ import React from 'react'
...
+    let browseProps = { assetType: assetType, section: section }
+    !section && delete browseProps.section
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx:14]: We could maybe type `browseProps` so we don't have to delete a prop (and maybe add more later

```ts
    const browseProps: BrowseOptions = { assetType: assetType }
    if(section) {
      browseProps.section = section
    }
```

and, this is just a suggestion, maybe we can call it browseOptions to follow the type

what do you think?

File: webapp/src/components/AssetPage/IconBadge/IconBadge.tsx [hunk 5]
```diff
@@ -2,18 +2,33 @@ import React from 'react'
...
 import { Props } from './IconBadge.types'
 import './IconBadge.css'
+import { Link } from 'react-router-dom'
 
-const IconBadge = ({ icon, text, onClick, className, children }: Props) => {
-  return (
-    <div className={classNames('IconBadge', className)} onClick={onClick}>
+const IconBadge = ({ icon, text, onClick, href, className, children }: Props) => {
+
+  const childrenInt = React.useMemo(() => (
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/IconBadge/IconBadge.tsx:9]: We could just import `useMemo` right?

```suggestion
  useMemo(() => {
```

File: webapp/src/components/AssetPage/IconBadge/IconBadge.tsx [hunk 6]
```diff
@@ -2,18 +2,33 @@ import React from 'react'
 import classNames from 'classnames'
 import { Props } from './IconBadge.types'
 import './IconBadge.css'
+import { Link } from 'react-router-dom'
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/IconBadge/IconBadge.tsx:5]: We have some [import order standards](https://www.notion.so/decentraland/dApps-Standards-760ee6551c9f4c00aa71f7d9c50a7646#8d48e37938124a41ac664112cb1e27f6) we should follow! (Sorry 👮‍♂️ )

File: webapp/src/components/AssetPage/IconBadge/IconBadge.tsx [hunk 7]
```diff
@@ -2,18 +2,33 @@ import React from 'react'
...
+        {childrenInt}
+      </Link>
+    )
+  } else {
+    return (
+      <div className={classNames('IconBadge', className)} onClick={onClick}>
+        {childrenInt}
+      </div>
+    )
+  }
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/IconBadge/IconBadge.tsx:31]: What do you think of the, maybe more succint:

```ts
  return href ? (
    <Link
      className={classNames('IconBadge', className)}
      onClick={onClick}
      to={href}
    >
      {childrenInt}
    </Link>
  ) : (
    <div className={classNames('IconBadge', className)} onClick={onClick}>
      {childrenInt}
    </div>
  )
```

my reasoning is that ternary operators is what we normally use inside JSX so it looks a bit more natural

File: webapp/src/components/AssetPage/SmartBadge/SmartBadge.container.ts [hunk 8]
```diff
@@ -1,25 +1,4 @@
...
-          assetType,
-          section: Section.WEARABLES,
-          onlySmart: true
-        })
-      )
-    )
-})
-
-export default connect(null, mapDispatch)(SmartBadge)
+export default connect()(SmartBadge)
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/SmartBadge/SmartBadge.container.ts:4]: Same thing as the [CategoryBadge.container.ts](https://github.com/decentraland/marketplace/pull/645/files#diff-9d2230b63c0fd2e01523e58a22e7ce5e5b4fd84dcc49433970d851924bed6934) with the index and this file!

File: webapp/src/components/AssetPage/SmartBadge/SmartBadge.tsx [hunk 9]
```diff
@@ -4,13 +4,22 @@ import { SmartIcon } from 'decentraland-ui'
 import IconBadge from '../IconBadge'
 import { Props } from './SmartBadge.types'
 import './SmartBadge.css'
+import { Section } from '../../../modules/vendor/decentraland'
+import { locations } from '../../../modules/routing/locations'
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/SmartBadge/SmartBadge.tsx:8]: We have some (annoying maybe :P) [import order standards](https://www.notion.so/decentraland/dApps-Standards-760ee6551c9f4c00aa71f7d9c50a7646#8d48e37938124a41ac664112cb1e27f6) we should follow! (super sorry lol)

```suggestion
import { Section } from '../../../modules/vendor/decentraland'
import { locations } from '../../../modules/routing/locations'
import IconBadge from '../IconBadge'
import { Props } from './SmartBadge.types'
import './SmartBadge.css'
```

File: webapp/src/components/AssetPage/SmartBadge/SmartBadge.tsx [hunk 10]
```diff
@@ -4,13 +4,22 @@ import { SmartIcon } from 'decentraland-ui'
...
+
+const SmartBadge = ({ assetType }: Props) => {
+  
+  const href = React.useMemo(() => locations.browse({
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/SmartBadge/SmartBadge.tsx:12]: We could just import `useMemo` right?

```suggestion
  useMemo(() => {
```

File: webapp/src/components/GenderBadge/GenderBadge.container.ts [hunk 11]
```diff
@@ -1,30 +1,4 @@
...
-            ? [WearableGender.MALE]
-            : [WearableGender.FEMALE]
-        })
-      )
-    )
-})
+import { connect } from 'react-redux'
 
-export default connect(null, mapDispatch)(GenderBadge)
+export default connect()(GenderBadge)
```
Comment (nicosantangelo) [inline webapp/src/components/GenderBadge/GenderBadge.container.ts:4]: Same thing as the [CategoryBadge.container.ts](https://github.com/decentraland/marketplace/pull/645/files#diff-9d2230b63c0fd2e01523e58a22e7ce5e5b4fd84dcc49433970d851924bed6934) with the index and this file!

File: webapp/src/components/GenderBadge/GenderBadge.tsx [hunk 12]
```diff
@@ -1,14 +1,26 @@
...
 import { Props } from './GenderBadge.types'
 import styles from './GenderBadge.module.css'
+import { locations } from '../../modules/routing/locations'
 
-const GenderBadge = ({ bodyShapes, withText, onClick }: Props) => {
+const GenderBadge = ({ bodyShapes, withText, assetType, section }: Props) => {
   const isUnisex = getIsUnisex(bodyShapes)
+  const isGenderBodyShape = isGender(bodyShapes, BodyShape.MALE)
+
+  const href = React.useMemo(() => locations.browse({
```
Comment (nicosantangelo) [inline webapp/src/components/GenderBadge/GenderBadge.tsx:15]: We could just import `useMemo` right?

```suggestion
  useMemo(() => {
```

File: webapp/src/components/GenderBadge/GenderBadge.tsx [hunk 13]
```diff
@@ -1,14 +1,26 @@
...
+import React from 'react'
+import { BodyShape, WearableGender } from '@dcl/schemas'
 import classNames from 'classnames'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
-import React from 'react'
 import { isGender, isUnisex as getIsUnisex } from '../../modules/nft/utils'
 import IconBadge from '../AssetPage/IconBadge'
 import { Props } from './GenderBadge.types'
 import styles from './GenderBadge.module.css'
+import { locations } from '../../modules/routing/locations'
```
Comment (nicosantangelo) [inline webapp/src/components/GenderBadge/GenderBadge.tsx:9]: We have some [import order standards](https://www.notion.so/decentraland/dApps-Standards-760ee6551c9f4c00aa71f7d9c50a7646#8d48e37938124a41ac664112cb1e27f6) we should follow! (😅 )

**Review by nicosantangelo — CHANGES_REQUESTED** [review]
Careful, I've noticed that some files like `CategoryBadge.tsx`, `SmartBadge.tsx` and `IconBadge.tsx` are not properly linted with prettier!

This looks super!

File: webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx (hunk 2 above)
Comment (braianj) [reply webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx:6]: Many thanks for remember me this

File: webapp/src/components/AssetPage/IconBadge/IconBadge.tsx (hunk 6 above)
Comment (braianj) [reply webapp/src/components/AssetPage/IconBadge/IconBadge.tsx:5]: 🙈

File: webapp/src/components/AssetPage/IconBadge/IconBadge.tsx (hunk 7 above)
Comment (braianj) [reply webapp/src/components/AssetPage/IconBadge/IconBadge.tsx:31]: I think the same

File: webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx [hunk 14]
```diff
@@ -1,14 +1,35 @@
-import React from 'react'
+import React, { useMemo, useEffect } from 'react'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { getSearchWearableSection } from '../../../modules/routing/search'
+import { locations } from '../../../modules/routing/locations'
 import IconBadge from '../IconBadge'
 import { Props } from './CategoryBadge.types'
+import { BrowseOptions } from '../../../modules/routing/types'
```
Comment (nicosantangelo) [inline webapp/src/components/AssetPage/CategoryBadge/CategoryBadge.tsx:7]: We should move this two lines up! as per standards 🙃

**Pull request merged**